src/appleseed/renderer/modeling/bssrdf/dipolebssrdf.h

```diff
@@ -241,7 +241,11 @@ void DipoleBSSRDF::do_prepare_inputs(
         values->m_precomputed.m_channel_pdf);
 
     // Precompute the (square of the) max radius.
-    values->m_precomputed.m_rmax2 = foundation::square(dipole_max_radius(foundation::min_value(values->m_precomputed.m_sigma_tr)));
+    values->m_precomputed.m_rmax2 =
+        foundation::square(
+            dipole_max_radius(
+                foundation::min_value(
+                    values->m_precomputed.m_sigma_tr)));
```

Comment:
Mmmh, from my perspective, precomputation optimisation came from the possibility to vectorize.

Where this:

```
for(value in values)
{
    values->m_precomputed.m_rmax2 =
foundation::square(
            dipole_max_radius(
                foundation::min_value(
                    values->m_precomputed.m_sigma_tr)));
}
```

Become this:
```
sigmas; // array of m_sigma_tr
rmax2; // array of m_rmax2
// both having the same size (if values.size() is known, you can allocate all those arrays at once)
for(const auto i : sigmas.size())
{
    rmax2[i] =
foundation::square(
            dipole_max_radius(
                foundation::min_value(sigmas[i])));
}
```

Note I have no idea what `values`, `m_precomputed` and `m_sigma_tr` are. I assume simple types here.

I have also no idea if it worth the cost (I have no numbers) but it looks like the precomputation process can be vectorization friendly without all those indirection.